packages/backend-plugin-api/src/services/definitions/lifecycleServiceRef.ts

```diff
@@ -0,0 +1,42 @@
+/*
+ * Copyright 2022 The Backstage Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import { createServiceRef } from '../system/types';
+
+/**
+ * @public
+ **/
+export type ShutdownHookOptions = {
+  fn: () => Promise<void>;
+};
+
+/**
+ * @public
+ **/
+export interface BackendLifecycle {
+  /**
+   * Register a function to be called and when the backend is shutting down.
+   */
+  addShutdownHook(options: ShutdownHookOptions): void;
```

Comment:
How would you like `on` based naming like events? `onShutdown`

Or even going all nuts, `on('shutdown', () => {}, { ... options })`, but maybe we'll want to add event specific arguments later on?